R/req-url.R

```diff
@@ -69,10 +69,10 @@ req_url_path_append <- function(req, ...) {
 
   url <- url_parse(req$url)
 
-  if (!grepl("^/", path) && (is.null(url$path) || !grepl("/$", url$path))) {
+  if (!grepl("^/", path)) {
     path <- paste0("/", path)
   }
 
-  url$path <- paste0(url$path, path)
+  url$path <- paste0(sub("/$", "", url$path), path)
```

Comment:
I'd move the `sub()` up to the block above so all the `/` handling stuff is in one place.

Reply:
Hmm, it's getting late around here, so I might not be getting exactly what you mean. Something like

```
url <- url_parse(req$url)

url$path <- sub("/$", "", url$path)

if (!grepl("^/", path)) {
  path <- paste0("/", path)
}

url$path <- paste0(url$path, path)
```
?

Reply:
Yeah, except I'd do:

```R
url$path <- sub("/$", "", url$path)
if (!grepl("^/", path)) {
  path <- paste0("/", path)
}
```

to make it more clear that it's a "paragraph" of related code